File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.swift
class Solution {
    func findMinArrowShots(_ points: [[Int]]) -> Int {
        var points = points.sorted {$0[0] < $1[0]}
        var shootArrow = 1
        var curr = points[0]
        for interval in points.dropFirst() {
            if interval[0] > curr[1] {
                shootArrow += 1
                curr = interval
            } else {
                curr = [min(curr[0], interval[0]), min(curr[1], interval[1])]
            }
        }
        return shootArrow
    }
}

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points):
        points.sort(key=lambda x: x[1])

        arrows = 0
        loc = -1
        for p in points:
            if loc != -1 and p[0] <= loc <= p[1]:
                continue
            arrows += 1
            loc = p[1]

        return arrows
